Library/Homebrew/test.rb

```diff
@@ -17,6 +17,15 @@
   ENV.extend(Stdenv)
   ENV.setup_build_environment
 
+  # This ensures all executables in test do are called with a full path, i.e.
+  # bin/"wget", or Formula["openssl"].opt_bin/"openssl" rather than just "wget".
+  # It's very likely to break tests, so phase it in via CI rather than user
```

Comment:
Counterpoint: who is running tests who isn't CI 😀. I think it's worth CI and local/non-developer test behaviour being identical so it allows people to debug failing tests locally (remember the pain from sandbox tests; we enabled sandbox testing for everyone much earlier because of that). 👍 from me to just enabling this everywhere. CC @ilovezfs in case you have strong opinions as the two of you are the ones who will likely feel the fallout from this.